### ISSUE TITLE
kVUID_BestPractices_StoreOpDontCareThenLoadOpLoad fails to check if the storeOp is VK_ATTACHMENT_STORE_OP_NONE

### ISSUE THREAD

**Issue by HuYuxin** [KhronosGroup/Vulkan-ValidationLayers#4533]
In this change https://github.com/KhronosGroup/Vulkan-ValidationLayers/commit/5eb5206e8c88a452d1be2219e05e09e032aa641a, we added a validation check to make sure if the current image usage is  `IMAGE_SUBRESOURCE_USAGE_BP::RENDER_PASS_READ_TO_TILE`, then the previous image usage should not be `IMAGE_SUBRESOURCE_USAGE_BP::RENDER_PASS_DISCARDED`.

And the way we determine if the image usage is `IMAGE_SUBRESOURCE_USAGE_BP::RENDER_PASS_DISCARDED` is below:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/master/layers/best_practices_utils.cpp#L2832
```
IMAGE_SUBRESOURCE_USAGE_BP usage = IMAGE_SUBRESOURCE_USAGE_BP::RENDER_PASS_DISCARDED;

if ((!FormatIsStencilOnly(attachment.format) && attachment.storeOp == VK_ATTACHMENT_STORE_OP_STORE) ||
                (FormatHasStencil(attachment.format) && attachment.stencilStoreOp == VK_ATTACHMENT_STORE_OP_STORE)) 
{
        usage = IMAGE_SUBRESOURCE_USAGE_BP::RENDER_PASS_STORED;
}
```
However if the storeOp is `VK_ATTACHMENT_STORE_OP_NONE`, the usage should not be `IMAGE_SUBRESOURCE_USAGE_BP::RENDER_PASS_DISCARDED` either.

We should probably added another enum on top of `RENDER_PASS_STORED` and `RENDER_PASS_DISCARDED`, to deal with the case when the storeOp is `VK_ATTACHMENT_STORE_OP_NONE`.

**Comment by paul-lunarg** [comment]
Indeed. It would make more sense for VK_ATTACHMENT_STORE_OP_NONE to not update the usage of the attachment, and keep the last usage, for example the load usage. I think creating a new usage wouldn't make sense as it's not really "used", but a NONE usage could be used to selectively not update the stored usage.

**Comment by paul-lunarg** [comment]
Actually, reading the documentation on STORE_OP_NONE :
> VK_ATTACHMENT_STORE_OP_NONE specifies the contents within the render area are not accessed by the store operation. However, if the attachment was written to during the render pass, the contents of the attachment will be undefined inside the render area.

This means that an attachment with store none can be undefined if present in the pColorAttachments or depth in a subpass that contains a draw operation. Even in this case we can't know for sure if it is drawn to.
For now I will fix this error triggering on NONE, as it is currently wrong.

**Comment by ShabbyX** [comment]
> This means that an attachment with store none can be undefined if present in the pColorAttachments or depth in a subpass that contains a draw operation. Even in this case we can't know for sure if it is drawn to.

You could potentially track color / depth / stencil masks + rasterizer discard + scissors etc to determine if a render pass modifies the contents of an attachment, but that would be approximate at best (like a shader may discard all fragments).

That said, the point of STORE_OP_NONE is that the application _promises_ not to write to the attachment. VVL might want to warn about when that's violated, but that's a completely different story from this warning.

In summary:

- LOAD_OP_NONE + STORE_OP_NONE means that the attachment is not accessed _at all_, so you could skip this render pass for the sake of storeOpXThenLoadOpY warnings
- LOAD_OP_LOAD + STORE_OP_NONE means the attachment was used read-only. Any warnings about storeOpXThenLoadOpLoad should already be generated for the beginning of this render pass, so not really necessary to generate it for the following render passes IMO (i.e. you can at this point stop considering the attachment discarded
- LOAD_OP_CLEAR + STORE_OP_NONE leads to undefined values because CLEAR is a write operation :)
- LOAD_OP_DONT_CARE + STORE_OP_NONE similarly leads to undefined values
